charmcraft/parts.py

```diff
@@ -184,6 +184,15 @@ def get_build_packages(self) -> Set[str]:
 
     def get_build_environment(self) -> Dict[str, str]:
         """Return a dictionary with the environment to use in the build step."""
+        try:
+            os_release = os_utils.OsRelease()
+            if (os_release.id(), os_release.version_id()) in (("centos", "7"), ("rhel", "7")):
+                # CentOS 7 Python 3.8 from SCL repo
+                # get_build_commands will ignored build.sh env
+                os.environ["PATH"] = "/opt/rh/rh-python38/root/usr/bin:" + os.environ["PATH"]
```

Comment:
I think it's a bad idea to "pollute" the "main" environment here; it's going to affect everything that comes after this.

If the need is to also use this path in `get_build_commands()` then I think a better way would be to also check for centos in that function and extend the PATH to use there; possibly with a helper function to isolate this try/except (or contextlib.supress).